doc/org/build/gameplay.md

```diff
@@ -0,0 +1,722 @@
+  snapToNavMesh = 1 << 6
+}
+
+// These values are booleans because they originate from an external source, like json in a gltf file.
+export interface WaypointParams {
+  canBeSpawnPoint: boolean;
+  canBeOccupied: boolean;
+  canBeClicked: boolean;
+  willDisableMotion: boolean;
+  willDisableTeleporting: boolean;
+  willMaintainInitialOrientation: boolean;
+  snapToNavMesh: boolean;
+}
+
+// When we inflate a waypoint component, we pack the booleans into the flags property
+export function inflateWaypoint(world: HubsWorld, eid: number, props: WaypointParams) {
+  addComponent(world, Waypoint, eid);
+  let flags = 0;
+  if (props.canBeSpawnPoint) flags |= WaypointFlags.canBeSpawnPoint;
+  if (props.canBeOccupied) flags |= WaypointFlags.canBeOccupied;
+  if (props.canBeClicked) flags |= WaypointFlags.canBeClicked;
+  if (props.willDisableMotion) flags |= WaypointFlags.willDisableMotion;
+  if (props.willDisableTeleporting) flags |= WaypointFlags.willDisableTeleporting;
+  if (props.willMaintainInitialOrientation) flags |= WaypointFlags.willMaintainInitialOrientation;
+  if (props.snapToNavMesh) flags |= WaypointFlags.snapToNavMesh;
+  Waypoint.flags[eid] = flags;
+
+  // More lines omitted
+}
+
+// Later, we can read the waypoint flags using bitwise &:
+const canBeSpawnPoint = Waypoint.flags[eid] & WaypointFlags.canBeSpawnPoint;
+```
+
+
+<a id="org58d595d"></a>
+
+## Tag components
+
+`bitECS` components with no properties are called tag components. It is useful to be able to tag an entity so that it appears in queries.
+
+
+<a id="orga5547d0"></a>
+
+## The escape hatch
+
+Sometimes, we need to store data that is just numbers and strings. Since we can&rsquo;t store the data in `bitECS` components, we store it in regular `Map` s instead.
+
+For example, the `MediaPDF` component stores a numeric `pageNumber`, and separately has a (uninspiringly named) `map` property:
+
+```typescript
+export const MediaPDF = defineComponent({
+  pageNumber: Types.ui8
+});
+MediaPDF.map = new Map();
+```
+
+In typescript, we specify the data types we will store in the map:
+
+```typescript
+export interface PDFResources {
+  pdf: PDFDocumentProxy;
+  material: MeshBasicMaterial;
+  canvasContext: CanvasRenderingContext2D;
+}
+export const PDFResourcesMap = (MediaPDF as any).map as Map<EntityID, PDFResources>;
+```
+
+It is our responsibility to clean up anything we put into the map:
+
+```typescript
+pdfExitQuery(world).forEach(function (eid) {
+  const resources = PDFResourcesMap.get(eid)!;
+  resources.pdf.cleanup();
+  disposeMaterial(resources.material);
+  PDFResourcesMap.delete(eid);
+});
+```
+
+
+<a id="orgf29a631"></a>
+
+## Associating entities with `Object3D` s
+
+We often associate an entity with an `Object3D`. We do this by adding an `Object3DTag` component to the entity, storing the association in `world.eid2obj`, and setting `obj.eid` to the `EntityID`.
+
+An entity can only be associated with a single `Object3D`.
```

Comment:
It would be good if we can describe why we want to avoid multiple Object3Ds per an entity. That means, what benefits this limitation that up to only a single Object3D can be associated with an entity give us?